api/src/graphql/connectors/index.js

```diff
@@ -1,47 +1,51 @@
 import _ from 'lodash';
 import uuidv4 from 'uuid/v4';
 
+function runQueryAndGetDataRecord(session, query, params) {
+  return session.run(query, params).then((result) => {
+    session.close();
+    if (!result.records) return null;
+    return result.records[0].toObject();
+  });
+}
```

Comment:
Ah right, however this function is for cases when we are not getting whole node and relationship objects, but rather data returned by cypher queries. I'm using it to return an array of data for the email service, but it could serve other functions in the future. Another example use of this function could be when you want to get a count of all people who realize Responsibilities for more than two different Needs. I could also get a count of all softDeleted nodes currently in the database, or the computation of graph density for a subgraph excluding deleted nodes and relationships.  

I could set the `__label ` property, but then it would be something generic like DataObject which would also be misleading. 

But this has me thinking that `runQueryAndGetDataRecord` is not the best name for it since I'm not getting a single record but some sort of calculation or set of fields and props. Would `runQueryAndGetData` be any clearer?

Reply:
Got it. How about runQueryAndGetRawData? That at least implies a bit that it hasn't been "processed". I'm still worried that someone will write a connector using it that returns nodes without setting the __label field. Maybe a comment about this in the code? Or some other pattern that makes this more clear? I don't have a suggestion for a super simple and elegant solution right now, so maybe a comment or two about it in the code is enough. Not worth getting stuck on :)

Reply:
runQueryAndGetRawData and a comment it is! :)